### ISSUE TITLE
Bug in searching of recurrent events (calendar)

### ISSUE THREAD

**Issue by feelwhy** [odoo/odoo#28288]
Impacted versions:
12.0 (both Community & Enterprise)

Steps to reproduce:
1. Generate **private** calendar events under 'User 1'
2. Open **tree view** of events under 'User 2'
3.  Click on any private recurrent event

Current behavior:
1. A user might see another user events
2. Error occurs. Look at the logs in the end of the issue
3. [Important for coding]. Search by IDs of virtual events doesn't work. E.g. it impossible to search by [('id', 'in', ['14-20181120060000', '15-20181116060000'])], while virtual events have only such IDs

Expected behavior:
1. A user might not see user events
2. No errors when clicking on events
3. [Important for coding]. Search by IDS should be processed in 'def search' of calendar.event not to make each time transformations to real_ids and back

Video/Screenshot link (optional):
```
Error:
Odoo Server Error
Traceback (most recent call last):
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/http.py", line 653, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/http.py", line 312, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/http.py", line 695, in dispatch
    result = self._call_function(**self.params)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/http.py", line 344, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/http.py", line 337, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/http.py", line 938, in __call__
    return self.method(*args, **kw)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/http.py", line 517, in response_wrap
    response = f(*args, **kw)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/web/controllers/main.py", line 962, in call_kw
    return self._call_kw(model, method, args, kwargs)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/web/controllers/main.py", line 954, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/api.py", line 749, in call_kw
    return _call_kw_multi(method, model, args, kwargs)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/api.py", line 736, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/calendar/models/calendar.py", line 1650, in read
    real_data = super(Meeting, real_events).read(fields=fields2, load=load)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/models.py", line 2757, in read
    self._read_from_database(stored, inherited)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/models.py", line 2902, in _read_from_database
    field.read(fetched)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/fields.py", line 2381, in read
    lines = comodel.search(domain, limit=self.limit)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/models.py", line 1553, in search
    res = self._search(args, offset=offset, limit=limit, order=order, count=count)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/calendar/models/mail_message.py", line 25, in _search
    return super(Message, self)._search(args, offset=offset, limit=limit, order=order, count=count, access_rights_uid=access_rights_uid)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/mail/models/mail_message.py", line 652, in _search
    allowed_ids = self._find_allowed_doc_ids(model_ids)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/mail/models/mail_message.py", line 582, in _find_allowed_doc_ids
    allowed_ids |= self._find_allowed_model_wise(doc_model, doc_dict)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/calendar/models/mail_message.py", line 33, in _find_allowed_model_wise
    return super(Message, self)._find_allowed_model_wise(doc_model, doc_dict)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/mail/models/mail_message.py", line 572, in _find_allowed_model_wise
    allowed_doc_ids = self.env[doc_model].with_context(active_test=False).search([('id', 'in', doc_ids)]).ids
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/models.py", line 1553, in search
    res = self._search(args, offset=offset, limit=limit, order=order, count=count)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/calendar/models/calendar.py", line 1763, in _search
    events = self.browse(events.get_recurrent_ids(args, order=order))
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/addons/calendar/models/calendar.py", line 1134, in get_recurrent_ids
    self._cr.execute(query, where_params_list)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/sql_db.py", line 148, in wrapper
    return f(self, *args, **kwargs)
  File "/home/feel/Odoo/Odoo12/odoo-12.0+e/odoo/sql_db.py", line 225, in execute
    res = self._obj.execute(query, params)
psycopg2.DataError: ОШИБКА:  неверное значение для целого числа: "15-20181031114120"
LINE 1: SELECT id, ("calendar_event"."id" in ('15-20181031114120','1...
                                              ^

```

**Comment by sswapnesh** [comment]
Related to (Can be considered duplicate as both refers to same piece of code and trace back) https://github.com/odoo/odoo/issues/27474

**Comment by zartashbaig** [comment]
This will fix this bug.
https://github.com/odoo/odoo/pull/28621/commits/76a17bb08efd1dd171d823a2e3323ab93b92a25e

**Comment by feelwhy** [comment]
@Yenthe666 yes, the commit by @zartashbaig solves the issue on my clean Odoo 12 enterprise.
Should I close the issue or wait until it is merged to 12.0?

**Comment by mart-e** [comment]
@Yenthe666 because it has not been merged in any branch, it is a dev commit. It should be reviewed first to unsure it is correct.
